src/Microsoft.Extensions.Options/OptionsService.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of IOptionsFactory.
+    /// </summary>
+    /// <typeparam name="TOptions">The type of options being requested.</typeparam>
+    public class OptionsService<TOptions> : IOptionsService<TOptions> where TOptions : class, new()
```

Comment:
How about something like:

```
interface IOptionsService<TOptions>
{
    TOptions Get(...)
}

class OptionsService<TOptions> : IOptionsService<TOptions> 
{
    // ....
}

interface ICachedOptionsService<TOptions> : IOptionsService<TOptions>
{
    // adding cache required stuff (e.g. reset, add etc)
}

class CachedOptionsService<TOptions> : ICachedOptionsService<TOptions>
{
    ctor(IOptionsService<TOptions> commonService) { // body }

    // cache related stuff (e.g. reset, add etc)
}
```

In that case we use different entities for different needs.

Thoughts?